src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/Testing/MsTestV2/TestMethodAttributeExecuteIntegration.cs

```diff
@@ -47,7 +51,7 @@ public class TestMethodAttributeExecuteIntegration
         /// <returns>A response value, in an async scenario will be T of Task of T</returns>
         public static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, CallTargetState state)
         {
-            if (Common.TestTracer.Settings.IsIntegrationEnabled(IntegrationId))
+            if (MsTestIntegration.IsEnabled)
             {
                 Scope scope = Common.TestTracer.ActiveScope;
```

Comment:
Should this be using `state.Scope`, since you are setting it in `OnMethodBegin`?

Reply:
Good catch, fixed in 45506f2